### DISCUSSION TITLE
The Operator Framework needs a Mission Statement

### DISCUSSION THREAD

**Discussion by awgreene** [operator-framework/community#97] [Announcements]
It's come to my attention that the Operator Framework has never formally released a mission statement.

The Operator Framework's mission statement should define why the organization exists, our core goals, and serve as a rallying point for all that we do.

I've created a Mission Statement to kick things off:

> The Operator Framework is committed to promoting the adoption of the Kubernetes Operator pattern by developing a suite of open-source tools that simplify the development, distribution, and management of Operators. Our mission is to create a world where the power and flexibility of the Operator pattern can be harnessed to automate any application, service, or workload.

**Comment by fgiloux** [comment]
> the Operator pattern can be harnessed to automate any application, service, or workload.

IMHO the operator pattern is for specific use cases and may not be suitable for any  application, service, or workload.

**Reply by awgreene** [reply]
Agreed, I generated this with CHAT GPT and called this out as something we ought to remove during our community meeting. Have any suggestions on how you'd like to change the wording?

**Reply by fgiloux** [reply]
giving it a try:
"Our mission is to create an ecosystem where the power and flexibility of the Operator pattern can be harnessed to make complicate systems easy to use"

**Comment by awgreene** [comment]
In today's community meeting, @bparees and @joelanford lead the discussion on whether or not the operator framework should be limited to managing the installation of operators. I would love to continue that discussion here.

**Reply by awgreene** [reply]
I personally believe that OLM V1 should support installing sets of manifests that represent a version of "a thing".

**Reply by bparees** [reply]
+1, opinion about the shape of that "Thing"(and special logic for defining/applying its resources) comes from the provisioner implementation, imho.

"Extensions" has been used elsewhere and is a reasonably neutral term that could be used in place of Operators, when talking about the thing that's being installed/managed.  Of course the Operator SDK is still going to be largely focused on helping implement true Operator(controller) patterns.  Though it might also help with the more generic activities of creating bundles, managing upgrade graphs, and such.

**Reply by fgiloux** [reply]
Not in disagreement but what do we intend to provide that is not already possible by largely adopted tools like helm, kustomize, ksonnet, skaffold, etc? What are the differentiators, the specific value adds for "things" that are not operators?

**Reply by bparees** [reply]
> what do we intend to provide that is not already possible by largely adopted tools like helm, kustomize, ksonnet, skaffold, etc? What are the differentiators, the specific value adds for "things" that are not operators?

dependency management and upgrade orchestration(including upgrade graph maintenance/enforcement) are the two that usually come to mind for me.  (non-operators can have dependencies and upgrades)

**Reply by fgiloux** [reply]
upgrade graph is a good one, although I believe it is more useful with complex systems, which is where operators shine. With helm, kustomize and others you can declare dependencies or compositions. Doing real dependency management with conflict prevention is for sure a bigger undertaking, which requires the control of a complete realm.
I was also thinking of the capability to provide a trusted supply chain, although helm can offer the same by storing charts as OCI bundles and it is bound to the bundle delivery mechanism.

**Reply by bparees** [reply]
helm's dependency management seems to be much more constrained to depending on a particular additional chart, whereas OLM is pursuing a more generic concept of "dependencies" (constraints/requirements is perhaps a better term)

**Reply by fgiloux** [reply]
I agree. My point is: where this more generic concept of "dependencies" will bring significant value, enough value that people may decide to replace something  that they have been using for years and are comfortable with even if it has little deficiencies here and there or start using another tool on top of it? My current feeling is that we will see that more with complex systems: multi-tenants, using CRDs, cluster scoped resources or having specific requirements like GPU, ARM-arch.
The target audience may not be as large as the one for helm and kustomize but these are systems driving a lot of very valuable workload, which in my opinion already makes a very good mission.
